tests/api2/test_306_chart_release.py

```diff
@@ -0,0 +1,191 @@
+    ipfs_version = list(results.json()['charts']['ipfs']['versions'].keys())[0]
+
+
+@pytest.mark.dependency(name='release_ipfs')
+def test_06_create_ipfs_chart_release(request):
+    depends(request, ['setup_kubernetes'], scope='session')
+    global release_id
+    payload = {
+        'catalog': 'OFFICIAL',
+        'item': 'ipfs',
+        'release_name': 'ipfs',
+        'train': 'charts',
+        'version': ipfs_version
+    }
+    results = POST('/chart/release/', payload)
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), int), results.text
+    job_status = wait_on_job(results.json(), 300)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+    release_id = job_status['results']['result']['id']
+    time.sleep(5)
+
+
+def test_07_get_ipfs_chart_release_catalog(request):
+    depends(request, ['release_ipfs'])
+    results = GET(f'/chart/release/id/{release_id}/')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert results.json()['catalog'] == 'OFFICIAL', results.text
+
+
+def test_08_get_ipfs_chart_release_catalog_train(request):
+    depends(request, ['release_ipfs'])
+    results = GET(f'/chart/release/id/{release_id}/')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert results.json()['catalog_train'] == 'charts', results.text
+
+
+def test_09_get_ipfs_chart_release_name(request):
+    depends(request, ['release_ipfs'])
+    results = GET(f'/chart/release/id/{release_id}/')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert results.json()['name'] == 'ipfs', results.text
+
+
+def test_10_get_chart_release_scaleable_resources():
+    results = GET('/chart/release/scaleable_resources/')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+
+
+@pytest.mark.dependency(name='used_ports')
+def test_11_get_chart_release_used_ports(request):
+    global port_list
+    results = GET('/chart/release/used_ports/')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), list), results.text
+    port_list = results.json()
+
+
+def test_12_verify_ipfs_chart_release_used_ports(request):
+    depends(request, ['release_ipfs', 'used_ports'])
+    results = GET(f'/chart/release/id/{release_id}/')
+    for port_dict in results.json()['used_ports']:
+        assert port_dict['port'] in port_list, results.text
+
+
+def test_13_get_ipfs_chart_release_upgrade_summary(request):
+    depends(request, ['release_ipfs'])
+    results = POST('/chart/release/upgrade_summary/', {'release_name': 'ipfs'})
+    assert results.status_code == 422, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert 'No update is available' in results.text, results.text
```

Comment:
It will probably be nice if we can have a custom catalog where we have 2 versions of a catalog item, and we can install the old one and see if we get an upgrade summary. I can easily set up the custom catalog if you would like me to :)

Reply:
Sounds good, we can also test rollback.